### ISSUE TITLE
Softmax ? DeepSeek Alt ?

### ISSUE THREAD

**Issue by fire17** [TheoLeeCJ/SemIf#17]
Hi there if I understood correctly...
You're using softmax at the end correct? 

If so can we take the lessons from recent DeepSeek and instead of softmax is there approximation step 

Or something things similar to make it even faster? 

Also how do we get blazing fast real time speeds over images (sub 100ms)

Thanks a lot and all the best!
❤️

**Comment by NicolaiLassen** [comment]
Hmm, to calibrate, one can do this: 

<img width="1710" height="1386" alt="Image" src="https://github.com/user-attachments/assets/63e088dd-b1de-4247-972d-9609c694961f" />

**Comment by NicolaiLassen** [comment]
Meaning: p = softmax(logits / T)
